internal/jobservice/application.go

```diff
@@ -0,0 +1,51 @@
+package jobservice
+
+import (
+	"sync"
+
+	"github.com/go-redis/redis"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/internal/common/auth/authorization"
+	grpcCommon "github.com/G-Research/armada/internal/common/grpc"
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+
+	"github.com/G-Research/armada/internal/jobservice/server"
+
+	"github.com/G-Research/armada/pkg/api/jobservice"
+)
+
+func StartUp(config *configuration.JobServiceConfiguration) (func(), *sync.WaitGroup) {
+	log.Info("Armada jobService service starting")
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	db := createRedisClient(&config.Redis)
+	defer func() {
+		if err := db.Close(); err != nil {
+			log.WithError(err).Error("failed to close Redis client")
+		}
+	}()
+
+	grpcServer := grpcCommon.CreateGrpcServer(config.Grpc.KeepaliveParams, config.Grpc.KeepaliveEnforcementPolicy, []authorization.AuthService{&authorization.AnonymousAuthService{}})
+	log.Info("JobService service listening on ", config.GrpcPort)
+
+	redisJobRepository := repository.NewRedisJobServiceRepository(db)
```

Comment:
Is Redis a good fit for this? Why do we need a RedisJobServiceRepository?

Reply:
I'd probably use postgres instead of Redis to store the cache state on disk. And then also store the state of all jobs watched in-memory in the application itself as a map.